test_files/return_this/return_this.ts

```diff
@@ -0,0 +1,21 @@
+class MethodsReturnThis {
+  returnsThis() {
+    return this;
+  }
+
+  explicitThis(): this {
+    return this;
+  }
+
+  templateAndThis<T>(t: T): this {
```

Comment:
Might be worth having an overloaded function too, to check the overload merging case.

Reply:
Done.